vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/FileArtifactHandler.java

```diff
@@ -257,7 +250,9 @@ public ImportInfoImpl accept(WorkspaceFilter wspFilter, Node parent,
                     path = path.substring(idx + 1);
                 }
                 // only update binary if import mode is not MERGE
-                if (wspFilter.getImportMode(parentNode.getPath()) != ImportMode.MERGE) {
+                // TODO: why? if not existing yet even with merge should be set afterwards
```

Comment:
@tripodsan Do you know why MERGE should not even set the property if it is not there yet?

Reply:
well, merge must not modify the property.  since the binary property is required on the file, it doesn't need to be processed here (i think :-)